conan.cmake

```diff
@@ -186,14 +186,22 @@ function(conan_cmake_install)
 endfunction()
 
 
+function(conan_cmake_setup_conanfile)
+  parse_arguments(${ARGV})
+  if(ARGUMENTS_CONANFILE)
+    # configure_file will make sure cmake re-runs when conanfile is updated
+    configure_file(${ARGUMENTS_CONANFILE} ${ARGUMENTS_CONANFILE}.junk)
```

Comment:
Hi, this is indeed some cmake-fu. From the documentation https://cmake.org/cmake/help/v3.0/command/configure_file.html:

> If the \<input\> file is modified the build system will re-run CMake to re-configure the file and generate the build system again.

So CMake will keep track of the input file if it changes, like it does with `include`'d .cmake files, by storing it in `${CMAKE_BINARY_DIR}/CMakeFiles/generate.stamp.depend`. However `configure_file` also requires an output file to be specified, but I remove it immediately to avoid confusion since we're not going to use it.